pkg/interface/weather/tile/tile.js

```diff
@@ -182,7 +178,91 @@ export default class WeatherTile extends Component {
     }
 
     if ('currently' in data && 'daily' in data) {
-      return this.renderWithData(data);
+      let weatherStyle = {
+        gradient1: "",
+        gradient2: "",
+        text: ""
+      };
+
+      switch(data.daily.icon) {
```

Comment:
Put this whole switch statement in its own function. General rule of thumb: one function shouldn't be more than 40 lines